src/Microsoft.VisualStudio.ProjectSystem.Managed/ProjectSystem/LanguageServices/ActiveConfiguredProjectsIgnoringTargetFrameworkProvider.cs

```diff
@@ -0,0 +1,200 @@
+        [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
+        internal async Task OnProjectFactoryCompletedAsync()
+        {
+            await InitializeCoreAsync(CancellationToken.None).ConfigureAwait(false);
+        }
+
+        protected override async Task InitializeCoreAsync(CancellationToken cancellationToken)
+        {
+            using (_tasksService.LoadedProject())
+            {
+                await GetActiveConfiguredProjectsMapAsync().ConfigureAwait(false);
+
+                // Listen to changes to "TargetFrameworks" property.
+                var watchedEvaluationRules = Empty.OrdinalIgnoreCaseStringSet.Add(ConfigurationGeneral.SchemaName);
+                _evaluationSubscriptionLink = _activeConfiguredProjectSubscriptionService.ProjectRuleSource.SourceBlock.LinkTo(
+                        new ActionBlock<IProjectVersionedValue<IProjectSubscriptionUpdate>>(OnProjectChanged),
+                        ruleNames: watchedEvaluationRules, suppressVersionOnlyUpdates: true);
+            }
+        }
+
+        private async Task OnProjectChanged(IProjectVersionedValue<IProjectSubscriptionUpdate> e)
+        {
+            IProjectChangeDescription projectChange = e.Value.ProjectChanges[ConfigurationGeneral.SchemaName];
+            if (projectChange.Difference.ChangedProperties.Contains(ConfigurationGeneral.TargetFrameworksProperty))
+            {
+                var targetFrameworks = projectChange.After.Properties[ConfigurationGeneral.TargetFrameworksProperty];
+                await GetActiveConfiguredProjectsMapAsync(targetFrameworks).ConfigureAwait(false);
+            }
+        }
+
+        protected override Task DisposeCoreAsync(bool initialized)
+        {
+            _evaluationSubscriptionLink?.Dispose();
+            return Task.CompletedTask;
+        }
+
+        /// <summary>
+        /// Gets the active configured projects for the given set of "TargetFrameworks" for a cross-targeting project.
+        /// </summary>
+        private async Task<ImmutableDictionary<string, ConfiguredProject>> GetActiveConfiguredProjectsMapAsync(string targetFrameworks)
+        {
+            lock (_gate)
+            {
+                if (string.Equals(targetFrameworks, _cachedTargetFrameworks, StringComparison.OrdinalIgnoreCase))
+                {
+                    return _cachedConfiguredProjectsMap;
+                }
+            }
+
+            var builder = ImmutableDictionary.CreateBuilder<string, ConfiguredProject>();
+            var knownConfigurations = await _services.ProjectConfigurationsService.GetKnownProjectConfigurationsAsync().ConfigureAwait(true);
+            var isCrossTarging = knownConfigurations.All(c => c.IsCrossTargeting());
+            var activeConfiguration = _services.ActiveConfiguredProjectProvider.ActiveProjectConfiguration;
+            bool resetCachedTargetFrameworks = false;
+
+            if (!activeConfiguration.Dimensions.Any())
```

Comment:
I think the issue might be that we need to mark this provider with capabilities ProjectConfigurationsInferredFromUsage - this will ensure the correct ProjectConfigurationsService has been MEF exported before this type is exported.